components/arm/eva/eva.go

```diff
@@ -130,6 +136,9 @@ func NewEva(ctx context.Context, cfg config.Component, logger golog.Logger) (arm
 }
 
 func (e *eva) JointPositions(ctx context.Context, extra map[string]interface{}) (*pb.JointPositions, error) {
+	if err := e.checkNotClosed(); err != nil {
```

Comment:
What if the arm gets closed after this check? Do we need to take a lock to prevent this?

Reply:
we could. the team should file a ticket if they want that

Reply:
This check doesn't seem useful without that. It looks like we're handling concurrency with closing, but we're not.

Reply:
I'm asking about the scenario where the arm gets closed after this check. I believe the remainder of the function relies on the arm not being closed.

Reply:
The close code ends up stopping the arm which should produce failures in progress based on how its op manager usage works. Code called after this would be caught by that. This just adds a further protection.

Reply:
Then why is this needed at all? It's an anti-pattern to check for a condition at the start of a function without taking a lock to prevent changes to that condition during the function's execution.

Reply:
I will remove it.